lib/network/connection_manager.go

```diff
@@ -194,9 +224,10 @@ func (c *ConnectionManager) Broadcast(message common.Message) {
 				}
 
 				if err != nil {
-					c.log.Error("failed to SendBallot", "error", err, "validator", v)
+					errs[v.Address()] = err
```

Comment:
> One way to solve this issue would be to pass a function to be executed on error instead.

@Geod24 
Could you let me know by an example?

Reply:
Define `Broadcaster.Broadcast` as:
```
func (b *SimpleBroadcaster) Broadcast(message common.Message, onError func(string, error))
```

And instead of `errs[v.Address()] = err` use: `onError(v.Address(), err)`
`onError` should be safe to use concurrently.